src/Components/v2/CarouselV2.tsx

```diff
@@ -0,0 +1,260 @@
+import { Box, ChevronIcon, Flex } from "@artsy/palette"
+import { Options as FlickityOptions } from "flickity"
+import React, { ReactNode, useRef, useState } from "react"
+import styled from "styled-components"
+import { left, LeftProps, right, RightProps } from "styled-system"
+import { useDidMount } from "Utils/hooks/useDidMount"
+import { Media } from "Utils/Responsive"
+
+type Arrow = (
+  props: {
+    Arrow: React.ReactType
+    getFlickity: any
+  } & Props
+) => React.ReactNode
+
+interface Props {
+  height?: number
+  data: object[] // This is designed to handle any shape of data passed, as long as its an array
+  render: (slide) => ReactNode
+  renderLeftArrow?: Arrow
+  renderRightArrow?: Arrow
+  onArrowClick?: () => void
+  options?: FlickityOptions
+}
+
+const isClient = typeof window !== "undefined"
+
+export class Carousel extends React.Component<Props> {
+  static defaultProps = {
+    height: 300,
+  }
+
+  render() {
+    return (
+      <>
+        {isClient && (
+          <Box key={Math.random()}>
+            <Media at="xs">
+              <SmallCarousel {...this.props} />
+            </Media>
+            <Media greaterThan="xs">
+              <LargeCarousel {...this.props} />
+            </Media>
+          </Box>
+        )}
+        {!isClient && (
+          <Box key={Math.random()}>
+            <CarouselServer {...this.props} />
+          </Box>
+        )}
+      </>
+    )
+  }
+}
+
+const renderSlides = props => {
+  return props.data.map((slide, index) => {
+    return <Box key={index}>{props.render(slide)}</Box>
+  })
+}
+
+const CarouselServer = (props: Props) => {
+  return (
+    <Flex
+      flexDirection="row"
+      position="relative"
+      justifyContent="space-around"
+      alignItems="center"
+      height={props.height}
+    >
+      <CarouselContainer height={props.height}>
+        <Flex>{renderSlides(props)}</Flex>
+      </CarouselContainer>
+    </Flex>
+  )
+}
+
+/*
+ * Returns null if no window for SSR
+ */
+const Flickity = isClient ? require("react-flickity-component") : () => null
+
+export const LargeCarousel = (props: Props) => {
+  const [currentSlideIndex, setSlideIndex] = useState(0)
+  const [lastItemVisible, setLastItemVisible] = useState(false)
+  const flicktyRef = useRef(null)
+
+  const isMounted = useDidMount()
+
+  let flickity = null
+
+  const checkLastItemVisible = () => {
+    if (!flickity.slides.length) {
+      return
+    }
+    const isLastItemVisible = flickity.selectedElements.includes(
+      flickity.getLastCell().element
+    )
+    setLastItemVisible(isLastItemVisible)
+  }
+
+  if (isMounted) {
```

Comment:
A little comment here to about what's going on / why this is needed; don't be afraid of comments 😄